ldap/servers/slapd/conntable.c

```diff
@@ -471,19 +502,32 @@ connection_table_move_connection_on_to_active_list(Connection_Table *ct, Connect
     connection_table_dump_active_connection(c);
 #endif
 
-    c->c_next = ct->c[0].c_next;
+    /* Map new connection to connection table list */
+    c->c_ct_list = connection_table_get_list(ct);
+
+    c->c_next = ct->c[c->c_ct_list][0].c_next;
     if (c->c_next != NULL) {
         c->c_next->c_prev = c;
     }
-    c->c_prev = &(ct->c[0]);
-    ct->c[0].c_next = c;
+    c->c_prev = &(ct->c[c->c_ct_list][0]);
+    ct->c[c->c_ct_list][0].c_next = c;
 
     PR_Unlock(ct->table_mutex);
 
 #ifdef FOR_DEBUGGING
     connection_table_dump_active_connections(ct);
 #endif
 }
+/* Get which active list to use */
+size_t
+connection_table_get_list(Connection_Table *ct)
```

Comment:
This function selects which listener thread will poll a new established connection.
It is based on modulo, so depending on connection closure we may end in listeners with unbalanced number of connections. Any reason you prefer this option rather that selecting a listener if its number of connection is low ?

Reply:
> Any reason you prefer this option rather that selecting a listener if its number of connection is low ?

I used a simple round robin method because I believe it would be faster than checking which list has the lowest number of connections. But, taking connection closure into consideration, we could end up with lists that are unbalanced. I like your idea, let me look into it.